plugins/flytekit-ray/flytekitplugins/ray/task.py

```diff
@@ -50,11 +57,31 @@ class RayFunctionTask(PythonFunctionTask):
     _RAY_TASK_TYPE = "ray"
 
     def __init__(self, task_config: RayJobConfig, task_function: Callable, **kwargs):
-        super().__init__(task_config=task_config, task_type=self._RAY_TASK_TYPE, task_function=task_function, **kwargs)
+        super().__init__(
+            task_config=task_config,
+            task_type=self._RAY_TASK_TYPE,
+            task_function=task_function,
+            **kwargs,
+        )
         self._task_config = task_config
 
     def pre_execute(self, user_params: ExecutionParameters) -> ExecutionParameters:
-        ray.init(address=self._task_config.address)
+        init_params = {"address": self._task_config.address}
+
+        ctx = FlyteContextManager.current_context()
+        if not ctx.execution_state.is_local_execution():
+            working_dir = os.getcwd()
+            init_params["runtime_env"] = {"working_dir": working_dir}
+
+            cfg = self._task_config
+            if cfg.excludes_working_dir:
+                init_params["runtime_env"]["excludes"] = cfg.excludes_working_dir
+
+            # fast register data with timestamp mtime=0 will be zipped and uploaded to ray gcs
+            # zip does not support timestamps before 1980 -> hacky workaround of touching all the files
```

Comment:
Can this comment say explicitly how the data is being moved around? My understanding is:

1. Flyte's fast register and sends it to the head node
2. Ray then zips the working directory and sends it to the ray gcs
3. Ray worker pulls the files from ray's gcs.

Is this correct?

Reply:
Its exactly this only with the addition that ray respects the given `excludes` variable (`list[str]`), which functions as a ignore file

Reply:
qq: could we exclude the file during the fast register? (.gitignore)
Do we have a case where we want to upload a file to the head node but don't want it on the worker nodes?"

Reply:
> Do we have a case where we want to upload a file to the head node but don't want it on the worker nodes?
No i dont think so.

What file are you trying to exclude? The fast register `.tar.gz`? Actually I am thinking, its only the .tar.gz file coming with the `mtime=0` right? If we exclude that we dont need to modify any timestamp on files?